src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -147,7 +148,13 @@ export class HttpConnection implements IConnection {
         }
 
         // Transport will not be null if state is connected
-        return this.transport!.send(data);
+        const send = () => this.transport!.send(data);
```

Comment:
Are there other places that needs to be fixed?

Reply:
I don't think so.

Reply:
I wonder if it wouldn't be better to put the data itself in a queue and create an async send loop that dequeues the data. If there's no data available, the loop could start waiting on a promise that gets tripped on the next send.

The benefit of this is that batching sends into a single post should be made much easier. I talked to @davidfowl about this, and I think we can just concatenate/append all the data and send it in a single call to transport.send which should result in a single post.

Reply:
That seems reasonable. What do I do to send a batch of data - send an array?

Reply:
Hrm. A send loop seems tricky to do right in the single-threaded JavaScript environment